src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -1448,8 +1470,11 @@ abstract class ManagedStrategy(
         return deterministicMethodDescriptor
     }
 
-    override fun onMethodCallReturn(descriptorId: Long, descriptor: Any?, receiver: Any?, params: Array<Any?>, result: Any?) {
+    override fun onMethodCallReturn(descriptorId: Long, descriptor: Any?, methodId: Int, receiver: Any?, params: Array<Any?>, result: Any?) {
         runInIgnoredSection {
+            // process intrinsic candidate methods which instantiate objects
+            processIntrinsicMethodEffects(methodId, result)
```

Comment:
Actually, it makes more sense to move the `if` check here, so it is obvious for the code reader that this processing is only applied to the intrinsic methods:
```kotlin
if (MethodIds.isIntrinsicMethod(methodId)) {
    processIntrinsicMethodEffects(methodId, result)
}
```

Reply:
also added a `check(...)` inside the `processIntrinsicMethodEffects`